service/src/main/java/com/commercetools/pspadapter/payone/IntegrationService.java

```diff
@@ -154,18 +168,35 @@ public int port() {
         return Integer.parseInt(systemProperty);
     }
 
-    private ImmutableMap<String, Object> createHealthResponse(@Nonnull final ServiceConfig serviceConfig) {
+    ImmutableMap<String, Object> createHealthResponse(@Nonnull final String version, @Nonnull final String title,
+                                                      List<TenantFactory> tenants) {
         final ImmutableMap<String, String> applicationInfo = ImmutableMap.of(
-                "version", serviceConfig.getApplicationVersion(),
-                "title", serviceConfig.getApplicationName());
-
-        final List<String> tenants = tenantFactories.stream()
-                .map(TenantFactory::getTenantName)
-                .collect(toList());
-
+                "version", version,
+                "title", title);
+        Map<String, CompletionStage<Integer>> tenantMap = checkTenantStatuses(tenants);
+        //resolve all completable stages
+        listOfFuturesToFutureOfList(new ArrayList<>(tenantMap.values()));
```

Comment:
we need to join here to block the execution and actually "resolve" the completable stages.. Otherwise, it the later line is not guaranteed to have resolved stages...

Reply:
086a9cc77